pkgs/development/python-modules/kaleido/default.nix

```diff
@@ -0,0 +1,77 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchurl,
+  autoPatchelfHook,
+  expat,
+  libGL,
+  nspr,
+  nss,
+  sqlite,
+  system ? builtins.currentSystem,
+}:
+
+buildPythonPackage rec {
+  pname = "kaleido";
+  version = "0.2.1.post1";
+  format = "wheel";
+
+  src =
+    let
+      pypiPackageName = "kaleido";
+    in
+    {
+      # This library is so cursed that I have to use fetchurl instead of fetchPypi. I am not happy.
+      x86_64-linux = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl";
+        hash = "sha256-qiHPG/HHj4+lCp99ReEAPDh709b+CnZ8+780S5W9w6g=";
+      };
+      aarch64-linux = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1.post1-py2.py3-none-manylinux2014_armv7l.whl";
+        hash = lib.fakeHash;
+      };
+      x86_64-darwin = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1-py2.py3-none-macosx_10_11_x86_64.whl";
+        hash = lib.fakeHash;
+      };
+      aarch64-darwin = fetchurl {
+        url = "https://files.pythonhosted.org/packages/py2.py3/k/kaleido/kaleido-0.2.1-py2.py3-none-macosx_11_0_arm64.whl";
+        hash = lib.fakeHash;
+      };
+    }
+    ."${system}" or (throw "Unsupported system for ${pname}: ${system}");
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [
```

Comment:
```suggestion
  buildInputs = [
    bash
```
It needs bash to modify shebang.
<summary>error log</summary><details>

```
❯ python
Python 3.12.5 (main, Aug  6 2024, 19:08:49) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import plotly.express as px
>>> fig = px.scatter(px.data.iris(), x="sepal_length", y="sepal_width", color="species")
>>> fig.write_image("figure.png", engine="kaleido")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/rjdw9s3lpvk6sfpncrj89xn7pa9q8mwm-python3-3.12.5-env/lib/python3.12/site-packages/plotly/basedatatypes.py", line 3835, in write_image
    return pio.write_image(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/rjdw9s3lpvk6sfpncrj89xn7pa9q8mwm-python3-3.12.5-env/lib/python3.12/site-packages/plotly/io/_kaleido.py", line 266, in write_image
    img_data = to_image(
               ^^^^^^^^^
  File "/nix/store/rjdw9s3lpvk6sfpncrj89xn7pa9q8mwm-python3-3.12.5-env/lib/python3.12/site-packages/plotly/io/_kaleido.py", line 143, in to_image
    img_bytes = scope.transform(
                ^^^^^^^^^^^^^^^^
  File "/nix/store/rjdw9s3lpvk6sfpncrj89xn7pa9q8mwm-python3-3.12.5-env/lib/python3.12/site-packages/kaleido/scopes/plotly.py", line 153, in transform
    response = self._perform_transform(
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/rjdw9s3lpvk6sfpncrj89xn7pa9q8mwm-python3-3.12.5-env/lib/python3.12/site-packages/kaleido/scopes/base.py", line 293, in _perform_transform
    self._ensure_kaleido()
  File "/nix/store/rjdw9s3lpvk6sfpncrj89xn7pa9q8mwm-python3-3.12.5-env/lib/python3.12/site-packages/kaleido/scopes/base.py", line 177, in _ensure_kaleido
    self._proc = subprocess.Popen(
                 ^^^^^^^^^^^^^^^^^
  File "/nix/store/pgb120fb7srbh418v4i2a70aq1w9dawd-python3-3.12.5/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/pgb120fb7srbh418v4i2a70aq1w9dawd-python3-3.12.5/lib/python3.12/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/nix/store/rjdw9s3lpvk6sfpncrj89xn7pa9q8mwm-python3-3.12.5-env/lib/python3.12/site-packages/kaleido/executable/kaleido'
```
</details>

I suggest running the examples in `passthru.tests` to see if the figures are generated.
https://github.com/plotly/Kaleido?tab=readme-ov-file#use-kaleido-to-export-plotlypy-figures-as-static-images

Reply:
I haven't used `passthru.tests` at all yet. Got any pointers?

Reply:
`nix-build -A python3Packages.kaleido.tests`

Reply:
Thanks! I'll look into it.

Reply:
Done!